### ISSUE TITLE
Can't set minZoom to 0

### ISSUE THREAD

**Issue by cyrstron** [istarkov/google-map-react#505]
This check in your code doesn't allow to set minZoom to 0. Because 0 is falsy value.

`return minZoom || DEFAULT_MIN_ZOOM;`

[https://github.com/istarkov/google-map-react/blob/master/src/google_map.js#L454](https://github.com/istarkov/google-map-react/blob/master/src/google_map.js#L454)

**Comment by itsmichaeldiego** [comment]
@cyrstron Great catch!

**Comment by btbensoft** [comment]
Same here **minZoom:0, maxZoom:4** in fact is 3 to 4

```
<GoogleMapReact
        bootstrapURLKeys={{ key: 'xxxxxxxxxxxxxxxx' }}
            onGoogleApiLoaded={({map, maps}) =>
                map.mapTypes.set("roadmap", new maps.ImageMapType({
                    getTileUrl: function (coord, zoom) {
                        return "http://localhost:5000/images/plans/1/7f8c7dea-a57c-4b95-88e6-008ba99476f7_" + zoom + "_" +  coord.x + "_" + coord.y + ".png"
                    },
                    tileSize:new maps.Size(256,256),
                    name:"GoogleRoadMaps",
                    maxZoom:4,
                    minZoom:0
                }))
            }
            yesIWantToUseGoogleMapApiInternals
            defaultZoom={1}
            defaultCenter={{ lat: 48.124039, lng: 15.8339266 }}
            zoom={0}>
        </GoogleMapReact>
```

Any idea when you will push this bug?

**Comment by thunderkid** [comment]
I've never managed to get zoom to go below 4. I've tried `minZoomOverride`, but that appears to do nothing. I've just also tried setting `minZoom` to `1` to see whether @cyrstron's falsey bug was the issue, but still it refuses to go below 4. As I recall @istarkov didn't have a need to zoom < 4 and thought that keeping high zoom would prevent the need to worry about duplicate markers.
Was zoom < 4 ever working for anyone?

**Comment by btbensoft** [comment]
@itsmichaeldiego any idea when you will fixe this issue? I need set minZoom to 0 because i got my own plan picture :S

**Comment by itsmichaeldiego** [comment]
@btbensoft TBH I am suuuper busy this days, its an easy fix so feel free to collaborate if you need it soon! I can promise I will merge it quick ;)

**Comment by btbensoft** [comment]
@itsmichaeldiego I found way to fix my problem : minZoom to **1** and set the property minZoomOverride to **true**

```
maxZoom: 4,
minZoom: 1,
minZoomOverride: true
```
but now i got an other problem, when the map are in "repeat" zoom 1 : the markers are not well positioned :

![Images](https://image.ibb.co/icwkXn/Capture3.png)

**Comment by itsmichaeldiego** [comment]
@btbensoft There is clearly a problem with the logic [here](https://github.com/istarkov/google-map-react/blob/master/src/google_map.js#L456-L461) I am taking a look right now.

**Comment by btbensoft** [comment]
@itsmichaeldiego yes you are right for the zoom 0 fix. 

But for the makers bug (when the map are repeat more that 2 time, see my exemple...). i think is more when you get the coordinate :  [here](
https://github.com/istarkov/google-map-react/blob/master/src/google_map_markers.js#206-L269) ?

```
const pt = this.props.geoService.project(
          latLng,
          this.props.projectFromLeftTop
        );
```

**Comment by itsmichaeldiego** [comment]
@btbensoft Oh yeah, you're right, I was just taking a look, please create an issue for this, I will take care of the zoom 0 fix first and then will jump to this one.

**Comment by itsmichaeldiego** [comment]
@btbensoft Just to clarify, to set a `minZoom`, its always required to set `overrideMinZoom`, so if it was not working without the `overrideMinZoom` value, then that makes sense. What would make sense here is that if we set a `minZoom`, its obvious that we want to overrideMinZoom.

**Comment by itsmichaeldiego** [comment]
@thunderkid @btbensoft @carsy PR open -> #525 to address this issue, feel free to review or make comments.

My decision is to get rid of `minZoomOverride` property, I think that if we set a `minZoom` is clear that we want to override it, let me know what you guys think.

**Comment by itsmichaeldiego** [comment]
@btbensoft The problem you're describing in https://github.com/istarkov/google-map-react/issues/505#issuecomment-371421838, I believe its related to https://github.com/istarkov/google-map-react/issues/421

**Comment by itsmichaeldiego** [comment]
@btbensoft Can you try adding the following css rules to your marker component?

```
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
```

**Comment by itsmichaeldiego** [comment]
@thunderkid @btbensoft @carsy Okey guys, this has been fixed and published in version `0.33`, but please pay double-attention to what it says [on the docs](https://github.com/istarkov/google-map-react/blob/master/API.md#override-the-default-minimum-zoom):

```
WARNING: Setting this option can break markers calculation, causing no homeomorphism between screen coordinates and map.
```

I will be working on this to see if there is a possibility to handle small zooms and marker's positions, but in the meantime, keep in mind that it might be an issue.
